engine-api/src/methods/get_balance.rs

```diff
@@ -45,11 +46,11 @@ async fn inner_execute(
     address: Address,
     block_number: BlockNumberOrTag,
     state_channel: mpsc::Sender<StateMessage>,
-) -> Result<U256, JsonRpcError> {
+) -> Result<Option<U256>, JsonRpcError> {
     let (tx, rx) = oneshot::channel();
-    let msg = Query::GetBalance {
+    let msg = Query::BalanceByHeight {
         address,
-        block_number,
+        height: block_number,
```

Comment:
We should just rename the `inner_execute` input to `height: BlockNumberOrTag`. Same applies to get_nonce.